packages/mediacenter/LibreELEC-settings/scripts/factory-reset

```diff
@@ -38,42 +32,37 @@ get_target() {
 }
 
 show_reset_msg() {
-  echo "RESET IN PROGRESS"
-  echo ""
-  echo "Please do not reboot or turn off your ${NAME} device!"
-  echo ""
+  echo -e "RESET IN PROGRESS\n"
+  echo -e "Please do not reboot or turn off your @DISTRONAME@ device!\n"
 }
 
-if [ -f /storage/.cache/reset_oe ] ; then
+if [ -f /storage/.cache/reset_oe ]; then
   # hard reset
   rm -f /storage/.cache/reset_oe
   get_target
-  if [ ! -z $target ] ; then
+  if [ -n $target ]; then
     show_reset_msg
 
     StartProgress spinner "Performing hard reset... "
-      umount /storage
-      mke2fs -t ext4 -m 0 $target &>/dev/null
-      if [ ! -z $label ] ; then
-        tune2fs -U random -L $label $target &>/dev/null
-      fi
-      if [ ! -z $uuid ] ; then
-        tune2fs -U $uuid $target &>/dev/null
-      fi
+      rm -rf /storage/* /storage/.* &>/dev/null
```

Comment:
IIRC the reason for the current approach is time; it takes a couple of seconds to nuke/pave the partition vs. much longer to recursively delete 3-4GB of files from a typical SD card or USB stick.

Reply:
I'd also prefer to keep the current mkfs approach, it's faster and safer - and also helps to get to a sane state eg after filesystem corruption.

It might be worth though to check if /storage actually is a block device and fall back to rm otherwise - that should allow hard reset to also work when netbooting with /storage on NFS.

BTW: using `/storage/.*` is dangerous, that'll match `.` and `..` as well (rm might prevent you from shooting yourself in the foot, but better don't rely on that).